packages/cli/src/commands/doctor/healthchecks/jdk.ts

```diff
@@ -0,0 +1,72 @@
+import {fetchToTemp} from '@react-native-community/cli-tools';
+import versionRanges from '../versionRanges';
+import {doesSoftwareNeedToBeFixed} from '../checkInstallation';
+import {logManualInstallation} from './common';
+import {HealthCheckInterface} from '../types';
+
+import {
+  setEnvironment,
+  updateEnvironment,
+} from '../../../tools/environmentVariables';
+import {join} from 'path';
+import {Ora} from 'ora';
+import {unzip} from '../../../tools/unzip';
+import {deleteFile} from '../../../tools/deleteFile';
+
+export default {
+  label: 'JDK',
+  getDiagnostics: async ({Languages}) => ({
+    needsToBeFixed: doesSoftwareNeedToBeFixed({
+      version:
+        typeof Languages.Java === 'string'
+          ? Languages.Java
+          : Languages.Java.version,
+      versionRange: versionRanges.JAVA,
+    }),
+
+    version:
+      typeof Languages.Java === 'string'
+        ? Languages.Java
+        : Languages.Java.version,
+    versionRange: versionRanges.JAVA,
+  }),
+  win32AutomaticFix: async ({loader}: {loader: Ora}) => {
+    try {
+      // Installing JDK 11 because later versions seem to cause issues with gradle at the moment
+      const installerUrl =
+        'https://download.java.net/java/GA/jdk11/9/GPL/openjdk-11.0.2_windows-x64_bin.zip';
+      const installPath = process.env.LOCALAPPDATA || ''; // The zip is in a folder `jdk-11.02` so it can be unzipped directly there
+
+      loader.start(
+        `Downloading JDK 11 from "${installerUrl}" (this may take a few minutes)`,
+      );
+
+      const installer = await fetchToTemp(installerUrl);
+
+      loader.text = `Installing JDK in "${installPath}"`;
+
+      await unzip(installer, installPath);
+
+      await deleteFile(installer);
```

Comment:
Can `unzip` throw an error? In such case it won't delete the file. Instead, we could add `finally` block to this try/catch and call delete there

Reply:
I'm going to pull some of the changes I'm doing in `install-android` here because I'm moving this around and  might as well take a look now.

Reply:
@thymikee I've moved things a bit around. The context is that I'm reusing this method to install Android Studio and the SDK as well. Can you please take another look?

Reply:
Specialized function makes sense. I left an additional comment on try/finally, maybe it wasn't so clear :D